moveit_core/profiler/include/moveit/profiler/profiler.h

```diff
@@ -128,41 +128,41 @@ class Profiler : private boost::noncopyable
   }
 
   /** \brief Destructor */
-  ~Profiler(void)
+  ~Profiler()
   {
     if (printOnDestroy_ && !data_.empty())
       status();
   }
 
   /** \brief Start counting time */
-  static void Start(void)
+  static void Start()  // NOLINT(readability-identifier-naming)
   {
     instance().start();
   }
 
   /** \brief Stop counting time */
-  static void Stop(void)
+  static void Stop()  // NOLINT(readability-identifier-naming)
   {
     instance().stop();
   }
 
   /** \brief Clear counted time and events */
-  static void Clear(void)
+  static void Clear()  // NOLINT(readability-identifier-naming)
   {
     instance().clear();
   }
 
   /** \brief Start counting time */
-  void start(void);
+  void start();
 
   /** \brief Stop counting time */
-  void stop(void);
+  void stop();
 
   /** \brief Clear counted time and events */
-  void clear(void);
+  void clear();
```

Comment:
[nit] Why do we have google style (that requires NOLINT because of our rules) naming and then pascal style names later on in the same file?  Was this copied from somewhere?

Reply:
This Profiler has the same member functions, both as static and non-static versions.
They are only distinguished by the capital letter.

Reply:
Is that a style we use in other places?  

I've just never seen it done that way and it would be confusing to me as a user of this that there are static and non-static versions of each and the implications of mixing them.  The doxygen comments don't explain which ones are static and how that differs.

It would be wrong but it would still build to do this, correct?
```c++
Profiler my_profiler;
my_profiler.Start();

...
my_profiler.stop();
```

Would a better API be making the static methods not part of the interface of the object and instead make them regular functions?

Reply:
I didn't want to enter a fundamental discussion here on function names. I just wanted to fix the broken CI for now. If you want to change API here, you should be aware that downstream projects might depend on it.
Apart from that, I found this API weird as well.

Reply:
Point taken on changing this being out of scope for this PR.  I only noticed it because of all the NOLINT comments and then when I looked closer I thought it seemed like a bad API.